build.sbt

```diff
@@ -0,0 +1,107 @@
+
+/* --------------------------------------------------------------------- */
+
+lazy val commonSettings: Seq[Setting[_]] = Metadata.settings ++ Seq(
+  releaseUseGlobalVersion := false,
+  crossScalaVersions      := scalaVersions,
+  scalaVersion            := scalaVersions.min,
+  pomIncludeRepository    := { _ => false },
+  javacOptions            ++= Seq("-source", "1.8", "-target", "1.8"),
+  scalacOptions           ++= Seq("-deprecation", "-unchecked"),
+  releasePublishArtifactsAction := PgpKeys.publishSigned.value,
+)
+
+lazy val clientSettings: Seq[Setting[_]] = Seq(
+  name                := Metadata.ghProject,
+  description         := "Scala client for the Guardian's Content API",
+  developers          := Metadata.clientDevs,
+  buildInfoKeys       := Seq[BuildInfoKey](version),
+  buildInfoPackage    := "com.gu.contentapi.buildinfo",
+  buildInfoObject     := "CapiBuildInfo",
+  releaseVersionFile  := file("version-client.sbt"),
+  libraryDependencies ++= clientDeps,
+  initialCommands in console := """
+    import com.gu.contentapi.client._
+    import com.gu.contentapi.client.model._
+    import scala.concurrent.ExecutionContext.Implicits.global
+    import scala.concurrent.Await
+    import scala.concurrent.duration._
+  """
+)
+
+lazy val defaultClientSettings: Seq[Setting[_]] = Seq(
+  name                := Metadata.ghProject + "-default",
+  description         := "Default scala client for the Guardian's Content API",
+  developers          := Metadata.clientDevs,
+  releaseVersionFile  := file("version-client.sbt"),
+  libraryDependencies ++= clientDeps ++ defaultClientDeps,
+  initialCommands in console := """
+    import com.gu.contentapi.client._
+    import com.gu.contentapi.client.model._
+    import scala.concurrent.ExecutionContext.Implicits.global
+    import scala.concurrent.Await
+    import scala.concurrent.duration._
+  """
+)
+
+lazy val awsSettings: Seq[Setting[_]] = Seq(
+  name                := Metadata.ghProject + "-aws",
+  description         := "AWS helper functionality for using Guardian's Content API scala client",
+  developers          := Metadata.awsDevs,
+  releaseVersionFile  := file("version-aws.sbt"),
+  libraryDependencies ++= awsDeps,
+)
+
+lazy val publishSettings: Seq[Setting[_]] = Seq(
+  resolvers += Resolver.sonatypeRepo("releases"),
+  publishTo := {
+    val nexus = "https://oss.sonatype.org/"
+    if (version.value.trim.endsWith("SNAPSHOT"))
+      Some("snapshots" at nexus + "content/repositories/snapshots")
+    else
+      Some("releases"  at nexus + "service/local/staging/deploy/maven2")
+  },
+  publishMavenStyle := true,
+  publishArtifact in Test := false,
+)
+
+lazy val releaseSteps: Seq[ReleaseStep] = Seq(
+  checkSnapshotDependencies,
+  inquireVersions,
+  runClean,
+  runTest,
+  setReleaseVersion,
+  commitReleaseVersion,
+  tagRelease,
+  publishArtifacts,
+  setNextVersion,
+  commitNextVersion,
+  releaseStepCommand("sonatypeReleaseAll"),
```

Comment:
Since this project is cross-versioned, I think that we need this step to be prepended with a `+`.

Reply:
That seems weird 🤔 I've never seen that before, do you have an example?

Reply:
Sorry, my mistake. I got confused between this and other steps. This should _not_ have a `+`!

Reply:
May I suggest using `sonatypeRelease` instead of `sonatypeReleaseAll`? The latter releases `all staging repositories`, therefore interfering with staging releases made by others in the same organisation.